loopchain/peer/block_manager.py

```diff
@@ -140,12 +142,28 @@ def _send_unconfirmed_block(self, block_: Block, target_reps_hash, round_: int):
             f"target_reps_hash({target_reps_hash})")
 
         block_dumped = self.blockchain.block_dumps(block_)
-        ObjectManager().channel_service.broadcast_scheduler.schedule_broadcast(
+        send_kwargs = {"block": block_dumped, "round_": round_, "channel": self.__channel_name}
+
+        release_recovery_mode = False
+        if conf.RECOVERY_MODE:
+            from loopchain.tools.recovery import Recovery
+            if self.blockchain.block_height <= Recovery.release_block_height():
+                util.logger.info(f"broadcast block({block_.header.height}) from recovery node")
+                send_kwargs.update({"from_recovery": True})
+
+            if self.blockchain.block_height >= Recovery.release_block_height():
+                release_recovery_mode = True
+
+        self.__channel_service.broadcast_scheduler.schedule_broadcast(
             "AnnounceUnconfirmedBlock",
-            loopchain_pb2.BlockSend(block=block_dumped, round_=round_, channel=self.__channel_name),
+            loopchain_pb2.BlockSend(**send_kwargs),
             reps_hash=target_reps_hash
         )
 
+        if release_recovery_mode:
+            conf.RECOVERY_MODE = False
+            util.logger.info(f"recovery mode released at {self.blockchain.block_height}")
```

Comment:
Why don't you move this code block to line 155?

Reply:
IMO, `RECOVERY_MODE` should be set disabled after generator broadcast unconfirmed block.